php/data/Database.php

```diff
@@ -13,21 +13,96 @@ public function __construct(){
         parent::__construct();
     }
 
-    // Test DB method
-    public function testDB(){
+    public function getUserInfo($returns, $column, $value){
 
-        $this->dbConnect();
+        // String of things we want to get from the database
+        // If $returns is 0, we pull everything
+        if(!$returns){
+            $select = "*";
+
+        // Otherwise, it contains something
+        }else {
+
+            // If it is an array...
+            if (is_array($returns)) {
 
-        $sql = "SELECT * FROM test";
+                // Implode the array in to a string, delimited with commas and surrounded with parentheses.
+                $select = "(" . implode(", ", $returns) . ")";
+            } else {
+                $select = "($returns)";
+            }
+        }
+
+        // Set up the sql query
+        $sql = "
+            SELECT $select FROM users WHERE $column = ?
+        ";
+
+        // Connect to the database
+        $this->dbConnect();
 
+        // Prepare, bind and execute the query
         $stmt = $this->con->prepare($sql);
+        $stmt->bindParam(1, $value);
         $stmt->execute();
         $stmt->setFetchMode(PDO::FETCH_ASSOC);
 
+        // Collect the results
         $results = $stmt->fetchAll();
 
+        // Disconnect from the DB
+        $this->dbDisconnect();
+
+        // Return the results
+        return($results);
+    }
+
+    public function registerAccount($username, $password, $email, $firstName, $lastName){
+
+        // Ensure that the email is an actual e-mail address
+        if(!filter_var($email, FILTER_VALIDATE_EMAIL)){
```

Comment:
This should never happen, an invalid mail should never be submitted. We can check before form submission